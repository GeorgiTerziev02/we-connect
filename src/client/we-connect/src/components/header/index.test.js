import React from 'react'
import renderer from 'react-test-renderer'
import Header from './index.js'
import TestingEnvironment from '../../test-utils/router'

describe('Header Component', () => {
    it('should render authenticated links', () => {
        const tree = renderer.create(
            <TestingEnvironment value={{
                user: {
                    id: 3123123
                },
                loggedIn: true
            }}>
                <Header />
            </TestingEnvironment>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    }),
        it('should render without authenticated links', () => {
            const tree = renderer.create(
                <TestingEnvironment value={{
                    loggedIn: false
                }}>
                    <Header />
                </TestingEnvironment>
            ).toJSON()
            expect(tree).toMatchSnapshot()
        })
})