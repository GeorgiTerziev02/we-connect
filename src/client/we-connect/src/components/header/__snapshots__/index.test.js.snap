// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render authenticated links 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark sticky-top fixed-top"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    <img
      alt=""
      src="https://res.cloudinary.com/dlgtvkmfb/image/upload/c_scale,h_36/v1594649995/weconnect_rmfzhf.png"
    />
  </a>
  <button
    aria-controls="responsive-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/"
        onClick={[Function]}
      >
        We Connect
      </a>
    </div>
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <form
        className="form-inline"
        onSubmit={[Function]}
      >
        <input
          className="mr-sm-2 form-control"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
        <button
          className="btn btn-outline-info"
          disabled={false}
          type="submit"
        >
          Search
        </button>
      </form>
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/share-post"
        onClick={[Function]}
      >
        Share
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/user/3123123"
        onClick={[Function]}
      >
        Profile
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/"
        onClick={[Function]}
      >
        Logout
      </a>
    </div>
  </div>
</nav>
`;

exports[`Header Component should render without authenticated links 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark sticky-top fixed-top"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    <img
      alt=""
      src="https://res.cloudinary.com/dlgtvkmfb/image/upload/c_scale,h_36/v1594649995/weconnect_rmfzhf.png"
    />
  </a>
  <button
    aria-controls="responsive-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/"
        onClick={[Function]}
      >
        We Connect
      </a>
    </div>
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
  </div>
</nav>
`;
